wren-lang/wren-cli
# pin versions

- reimplement Wren CLI as pure Wren
- add -e for eval
- move resolveModule into Wren
- move loadModule into Wren
- run stdin via -
- dynamic library loading
- (chore) move CLI source into cli folder
- remove old C module resolve/load code
- Path class for Resolver
- single CLI inc file
- cli/path.wren
- path: better support for windows
- mix in our path tests
- rename: wrenc makefiles
- rename CLI, readme, docs, etc.
- list what we added
- bundle essentials
- pre-requisites
- (chore) push releases to GitHub
- (fix) Stdin.readByte now propery removes single bytes from buffer
- (docs) add reflect module in readme
- (enh) Runtime module
- (enh) disallow compiling into an existing named module
- add runtime to makefiles
- declare extern
- (enh) support absolutely paths for scripts
- (enh) controlled crashes should not include CLI in stack trace
- (enh) add Process.exit() and Process.exit(_)
- (enh) Add `Stderr.write(_)` and `Stderr.print(_)`
- (fix) correct number of slots
- (fix) fix compiler warning
- (fix) Process.exit works now
- adds Process.exec(cmd, [args])
- (enh) Process.exec inherits stdout, stderr
- add Process.exec to README
- passing tests
- bump version
- allow multiple wren_modules paths to be searched
- declare S_IRWXU on Windows platforms
- CI & artifacts
- (ci) build mac
- (ci) build windows
- (fix) test: cwd on windows
- (fix) test: pid test
- (fix) mode test
- CI: dependencies/wren-essentials
- (chore) split process.exec tests between unix/windows
- artifacts then test
- ensure slots, even for slot 0
- (chore) windows CI: use debug build
- (fix) proper number of slots, release fiber handle in case of err
- (fix) free resolver VM
- (fix) do not free stdinStream, let uvShutdown do it
- abstract path unit tests
- (fix) Use Path for dirname resolution
- fixup: correct name of asset
- bump: 0.2.90 [no ci]
- (chore) update release script [no ci]
- release 0.2.90
- do not forget wren-package
- build windows release
- add installation/homebrew instructions
- update build badge
- do not need a large heap for resolver
- nicer errors if the resolver VM blows up
- easily compile in additional modules
- build in wren-package
- release 0.2.91
- add sentinel
- remove wren-package external dependency
- should not need to build either [skip ci]
- re-release 0.2.91
- bump badges
- (enh) Add `Process.chdir`, exposing libuv `uv_chdir` (#8)
- (docs) restructure usage examples (#9)
- (fix) use absolute paths for module directory resolution
- (chore) update build files
- (docs) update list of modules/functions
- (chore) regenerate all build files
- (enh) Runtime.assertVersion (#15)
- (env) add Ensure module for argument validation (#13)
- (fix) Path normalize `./../` works corretly now (#14)
- Dirname and basename (#16)
- simplify tests further
- add better capability detection
- prepare for 0.3 release
- credit
- (enh) ctrl-H / backspace handling (#19)
- bump v0.3 changelog
- bump version in runtime info
- bump runtime version info
- bump runtime tests
- (chore) pin wren-essentials version for CI and releases
